smartapp-core/src/main/java/com/smartthings/sdk/smartapp/core/extensions/HttpVerificationService.java

```diff
@@ -55,29 +56,33 @@ public HttpVerificationService(String publicKeyPath) {
         InputStream in = HttpVerificationService.class.getResourceAsStream(publicKeyPath);
         if (in == null) {
             // This is not an error until we are past the PING step.
-            LOG.info("no public key yet; will only accept PING lifecycle requests");
+            log.info("no public key yet; will only accept PING lifecycle requests");
             publicKey = null;
             return;
         }
 
-        LOG.debug("Looking for public key file: " + publicKeyPath);
+        if (log.isDebugEnabled()) {
+            log.debug("Looking for public key file: " + publicKeyPath);
+        }
         try (BufferedReader reader = new BufferedReader(new InputStreamReader(in, "UTF-8"))) {
             publicKey = reader.lines().collect(Collectors.joining("\n"));
         } catch (IOException ioException) {
-            LOG.error("failed to read " + publicKeyPath + " file");
+            if (log.isErrorEnabled()) {
+                log.error("failed to read " + publicKeyPath + " file");
+            }
             throw new RuntimeException("failed to read " + publicKeyPath + " file", ioException);
         }
     }
 
     public boolean verify(String method, String uri, Map<String, String> headers) {
         if (publicKey == null) {
-            LOG.error("Public key file not set up properly. Please see README (Configure Public Key) for"
+            log.error("Public key file not set up properly. Please see README (Configure Public Key) for"
                     + " directions and don't forget to restart this server.");
             return false;
         }
         boolean verified = verifyRequest(method, uri, headers);
         if (!verified) {
-            LOG.error("Request not verified!");
+            log.error("Request not verified!");
```

Comment:
is there a reason not to check `if (log.isErrorEnabled()) {` is some places but not others?

Reply:
Yeah, you want to check when the interpreter has to do work just to make the string that gets passed in (like concatenating a string and a variable but not two strings which the compiler would concatenate).

I also never bother at log levels that I expect to be pretty much always logged anyway.

[PMD](https://pmd.github.io/) has a rule that enforces the checks and I have it enabled only for trace, debug and info.

Reply:
makes sense